python/pyspark/pandas/spark/functions.py

```diff
@@ -32,6 +41,22 @@ def repeat(col: Column, n: Union[int, Column]) -> Column:
     return _call_udf(sc, "repeat", _to_java_column(col), n)
 
 
+def lit(literal: Any) -> Column:
+    """
+    Creates a Column of literal value.
+    """
+    if isinstance(literal, np.int64):
```

Comment:
@xinrong-databricks, I think we can just support this natively all across PySpark. Can you add an input converter here (https://github.com/apache/spark/blob/20750a3f9e13a2f02860859f87bbc38a18cba85e/python/pyspark/sql/types.py#L1608-L1622) and see if it works?

Also, I think we can simplify it by `item()` (https://stackoverflow.com/a/11389998/2438480) with  `np.generic` type check (https://numpy.org/doc/stable/reference/arrays.scalars.html).

cc @ueshin @itholic too

Reply:
cc @mengxr too FYI

Reply:
Certainly, I am working on this.

Reply:
Adding an input converter might be tricky in this case.

Py4j takes the instance of `java.lang.Integer` as `int` rather than a `JavaObject`, but the return value of `def convert` is expected to be a `JavaObject`.

Unfortunately, there is also no wrapper in `java.sql` package for numeric values.

Do you have insights on this?

Attached please find my pseudo-code
```py
class SomeConverter(object):
    def can_convert(self, obj):
        import numpy as np
        return isinstance(obj, np.generic)

    def convert(self, obj, gateway_client):
        Integer = JavaClass("java.lang.Integer", gateway_client)
        return Integer.valueOf(obj.item())
```

And the exception stack trace looks like
```
Traceback (most recent call last):
  File "/Users/xinrong.meng/spark/python/pyspark/pandas/tests/data_type_ops/test_udt_ops.py", line 43, in test
    print(F.lit(np.int64(1)))
  File "/Users/xinrong.meng/spark/python/pyspark/sql/functions.py", line 100, in lit
    return col if isinstance(col, Column) else _invoke_function("lit", col)
  File "/Users/xinrong.meng/spark/python/pyspark/sql/functions.py", line 58, in _invoke_function
    return Column(jf(*args))
  File "/miniconda2/envs/pyspark-dev-pd-1.1.5/lib/python3.9/site-packages/py4j/java_gateway.py", line 1313, in __call__
    temp_arg._detach()
AttributeError: 'int' object has no attribute '_detach'
```

Reply:
Just found something: https://www.py4j.org/advanced_topics.html#boxing

Reply:
I am wondering if we may support creating a Column of numpy literal value in pandas-on-Spark first. We might need more research on supporting that in PySpark.

Reply:
I'm fine with the current implementation since sounds like the converter is difficult.
I'd leave this to @HyukjinKwon.

Reply:
One workaround might be to define classes that inherit these classes e.g.) `class Py4JInteger extends java.lang.Integer` in our codebase, and use it for Java references ..

Okay but I am fine as is too about this.

Reply:
Okie, I am fine as is.

Reply:
Thank you!